authentication/pipeline/user.py

```diff
@@ -190,10 +222,11 @@ def require_micromasters_provider(
     user_storage = strategy.storage.user
 
     # if the user only has one social auth and it is MicroMasters, prompt them to login via that method
-    if (
-            user_storage.get_social_auth_for_user(user, provider=MicroMastersAuth.name).exists() and
-            user_storage.get_social_auth_for_user(user).count() == 1
-    ):
-        raise RequireProviderException(backend, MicroMastersAuth.name)
+    social_auths = user_storage.get_social_auth_for_user(user)
+    if len(social_auths) != 1:
+        return {}
 
+    social_auth = social_auths[0]
+    if social_auth.provider == MicroMastersAuth.name:
```

Comment:
You're right that this isn't related to the issue for this PR. I was tinkering with this step early on and this was sort of a leftover from that. I'm not sure that it reintroduces the race condition though. `len(social_auths)` runs the query, and any subsequent access to the `social_auths` variable uses Django's cache instead of running another query. `social_auths` could be wrapped in a `list()` call to make that more explicit. In short, there's only one query being run here, so no chance of a race condition that I can spot. More generally, I think this is more efficient. I would definitely expect one query that fetches no more than 3 objects to be faster than 2 separate queries.

Reply:
You make a good point. We can leave it for now since you've already written it, but I think it's generally important to partition unrelated changes into separate commits/PRs.